coderd/userauth.go

```diff
@@ -579,6 +580,28 @@ func (api *API) loginRequest(ctx context.Context, rw http.ResponseWriter, req co
 			})
 			return user, rbac.Subject{}, false
 		}
+
+		af := map[string]string{
+			"automatic_actor":     "coder",
+			"automatic_subsystem": "dormancy",
+		}
+
+		wriBytes, err := json.Marshal(af)
+		if err != nil {
+			api.Logger.Error(ctx, "marshal additional fields for dormancy audit", slog.Error(err))
+			wriBytes = []byte("{}")
+		}
+
+		audit.BackgroundAudit(ctx, &audit.BackgroundAuditParams[database.User]{
```

Comment:
This doesn't audit if we fail to update the user status. Also why is this a background audit if it's part of an API request?

Reply:
Personally I don't really think we get anything valuable from generating an audit log if this fails. The main reason for failed audit logs is for logging users attempting to take an action for which they don't have permission, or similar. The status failing to update here doesn't really matter, it can only ever be a 500 error. Plus, it'll be captured by the main login audit anyways.

I think `BackgroundAudit` is poorly named, it's really just for anything not specifically tied to a request. I don't think it make sense to use a request audit here because the status could successfully be changed while the api request as a whole fails. Also, since dormancy is also a background task, it maintains consistency. I think having two request audits would just make it more confusing.